4337/contracts/EIP4337Module.sol

```diff
@@ -59,6 +61,21 @@ abstract contract EIP4337Module is HandlerContext, CompatibilityFallbackHandler
         }
     }
 
+    /// @notice Executes user operation provided by the entry point
+    /// @dev Reverts if unsuccessful
+    /// @param executionData Execution data. Expects it to be a call to the `execTransactionFromModule` function of the Safe.
+    ///                      Validated in the `validateUserOp` function.
+    function executeUserOp(bytes calldata executionData) external {
+        address entryPoint = _msgSender();
+        require(entryPoint == supportedEntryPoint, "Unsupported entry point");
+
+        (bool outerCallSuccess, bytes memory returnData) = msg.sender.call(executionData);
```

Comment:
why not use `execTransactionFromModule` directly, as we do in the validate function. This way we do not need to do any manual decoding here and asume types.

I assume that there would be some gas overhead right? If it is about this, then I would at least check the first 4 bytes to avoid that it calls another function